packages/firestore/src/util/misc.ts

```diff
@@ -28,8 +29,17 @@ export class AutoId {
     const chars =
       'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
     let autoId = '';
-    for (let i = 0; i < 20; i++) {
-      autoId += chars.charAt(Math.floor(Math.random() * chars.length));
+    while (autoId.length < 20) {
+      const bytes = PlatformSupport.getPlatform().randomBytes(40);
+      for (const b of Array.from(bytes)) {
```

Comment:
Nit: You can save one copy here if you do a `for(let i = 0; i < bytes.length; ++i)` loop.

Reply:
Done.